src/health-check.js

```diff
@@ -1,25 +1,77 @@
 
 import { serializeError } from './logging.js'
+import { telemetry } from './telemetry.js'
 
-async function checkReadiness ({ awsClient, readiness, logger }) {
-  try {
-    await Promise.all([
-      awsClient.dynamoQueryBySortKey({
-        table: readiness.dynamo.table,
-        keyName: readiness.dynamo.keyName,
-        keyValue: readiness.dynamo.keyValue
-      }),
-      awsClient.s3Fetch({
-        region: readiness.s3.region,
-        bucket: readiness.s3.bucket,
-        key: readiness.s3.key
-      })
-    ])
-    return 200
-  } catch (err) {
-    logger.error({ err: serializeError(err) }, 'Readiness Probe Failed')
-    return 503
+const RESET_COUNTER = 1e6
+const SUCCESS_CODE = 200
+const ERROR_CODE = 503
+
+const checks = [
+  {
+    name: 'dynamo',
+    fn: async ({ awsClient, readiness }) => {
+      try {
+        await awsClient.dynamoQueryBySortKey({
+          table: readiness.dynamo.table,
+          keyName: readiness.dynamo.keyName,
+          keyValue: readiness.dynamo.keyValue
+        })
+      } catch (err) {
+        return err
+      }
+    },
+    counter: 0
+  },
+  {
+    name: 's3',
+    fn: async ({ awsClient, readiness }) => {
+      try {
+        await awsClient.s3Fetch({
+          region: readiness.s3.region,
+          bucket: readiness.s3.bucket,
+          key: readiness.s3.key
+        })
+      } catch (err) {
+        return err
+      }
+    },
+    counter: 0
+  }]
+const CHECKS_LENGHT = checks.length
+
+let totalCounter = 0
+async function checkReadiness ({ awsClient, readiness, samplings = [], logger }) {
+  const index = totalCounter % CHECKS_LENGHT
+  const check = checks[index]
+  const sampling = samplings[index] ?? 1
+  let code
```

Comment:
Why is this tied to the index of the set of checks?  It seems like we would only ever do a single check in any readiness check in this case, but if I specify samplings=[1,1] we would effectively toggle between checks and not run both checks every call of the readiness probe.

Would this be simpler if we for checks.forEach(...) and let each check's counter decide if it should be fired during this check?